test/field/detail/CMSMapField.i.hh

```diff
@@ -0,0 +1,68 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file CMSMapField.i.hh
+//---------------------------------------------------------------------------//
+#include "CMSMapField.hh"
+
+#include "base/Macros.hh"
+#include "base/Types.hh"
+#include "base/Units.hh"
+
+#include <math.h>
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with the shared magnetic field map data (FieldMapGroup).
+ */
+CELER_FUNCTION
+CMSMapField::CMSMapField(const FieldMapRef& shared) : shared_(shared) {}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Retrieve the magnetic field value for the given position.
+ */
+CELER_FUNCTION Real3 CMSMapField::operator()(Real3 pos) const
+{
+    CELER_ENSURE(shared_);
+
+    Real3 value{0, 0, 0};
+
+    real_type r = std::sqrt(pos[0] * pos[0] + pos[1] * pos[1]);
+    real_type z = pos[2];
+
+    size_type ir = (size_type)(r);
+    size_type iz = (size_type)(z + shared_.params.offset_z);
```

Comment:
`static_cast` rather than C-style for these fellas

Reply:
Done.